command.go

```diff
@@ -1119,7 +1122,13 @@ func (c *Command) ExecuteC() (cmd *Command, err error) {
 		// Always show help if requested, even if SilenceErrors is in
 		// effect
 		if errors.Is(err, flag.ErrHelp) {
-			cmd.HelpFunc()(cmd, args)
+			// The call to execute() above has parsed the flags.
+			// We therefore only pass the remaining arguments to the help function.
+			argWoFlags := cmd.Flags().Args()
+			if cmd.DisableFlagParsing {
+				argWoFlags = flags
+			}
+			cmd.HelpFunc()(cmd, argWoFlags)
```

Comment:
I wanted to discuss backwards-compatibility for this change.
Before this PR cobra would pass all the arguments on the command-line (which was incorrect).
With this PR cobra will pass only the arguments after removing the sub-command and flags, as it does when calling the `*Run/RunE` functions.

It is possible that some programs worked around this bug and fixing it may affect them.

However, considering that before this PR the help function sometime received all the args (when using the `--help` flag) but no args at all when called from the `help` command, it makes it less likely that a workaround used the actual `args` parameter.

So, I feel it is ok to make this change but if we do maybe we should include a note in the release notes.

@jpmcb how do you feel about this?

Reply:
Your use case description is correct.  However, there is another (obscure) situation, which is actually the one I'm really trying to fix 😄.

The `tanzu` CLI uses plugins to provide many of its sub-commands.  So a user may do:
`tanzu help cluster list`
to get the help on the `cluster list` command.
However, the `tanzu` CLI is only has `cluster` as a sub-command and needs to call that "cluster" plugin to ask it for the help for the `list` command.  Therefore, the arguments passed to the original `tanzu help` command are required so which which sub-command of the plugin the user is asking about.

Reply:
Currently, the `tanzu` CLI code has to use `os.Args` to get that info: https://github.com/vmware-tanzu/tanzu-cli/blob/6a11ce93cd4e811e213e8439e090e1d73a053fd3/pkg/cli/plugin_cmd.go#L192-L201